HpToolsLauncher/Runners/AlmTestSetsRunner.cs

```diff
@@ -1131,318 +1444,413 @@ private TestRunResults UpdateTestStatus(TestSuiteRunResults runResults, ITestSet
+                        }
+                        else if (enmState != TestState.Waiting)
+                        {
+                            ConsoleWriter.WriteLine(string.Format(Resources.AlmRunnerStatWithMessage, activeTestDesc.TestName, testExecStatusObj.TSTestId, statusString, testExecStatusObj.Message));
+                        }
+                        if (System.IO.File.Exists(abortFilename))
+                        {
+
+                            scheduler.Stop(currentTestSetInstances);
+                            //stop working 
+                            Environment.Exit((int)Launcher.ExitCodeEnum.Aborted);
+                            break;
+                        }
                     }
                 }
-                catch
+
+                //wait 0.2 seconds
+                Thread.Sleep(200);
+
+                //check for abortion
+                if (System.IO.File.Exists(abortFilename))
                 {
-                    retVal = "";
+                    _blnRunCancelled = true;
+
+                    ConsoleWriter.WriteLine(Resources.GeneralStopAborted);
+
+                    //stop all test instances in this testSet.
+                    scheduler.Stop(currentTestSetInstances);
+
+                    ConsoleWriter.WriteLine(Resources.GeneralAbortedByUser);
+
+                    //stop working 
+                    Environment.Exit((int)Launcher.ExitCodeEnum.Aborted);
                 }
             }
-            retVal = ConsoleWriter.FilterXmlProblematicChars(retVal);
-            return retVal;
         }
 
         /// <summary>
-        /// checks Qc version (used for link format, 10 and smaller is old) 
+        /// gets a link string for the test run in Qc
         /// </summary>
-        /// <returns>true if this QC is an old one</returns>
-        private bool CheckIsOldQc()
+        /// <param name="prevTest"></param>
+        /// <param name="runId"></param>
+        /// <returns></returns>
+        private string GetTestRunLink(ITSTest prevTest, int runId)
         {
-            string ver = null;
-            int intver = -1;
-            string build = null;
-            TdConnection.GetTDVersion(out ver, out build);
-            bool oldQc = false;
-            if (ver != null)
-            {
-                int.TryParse(ver, out intver);
-                if (intver <= 10)
-                    oldQc = true;
-            }
-            else
+            var oldQc = CheckIsOldQc();
+            var useSsl = MQcServer.Contains("https://");
+
+            string linkStr = "";
+            if (!oldQc)
             {
-                oldQc = true;
+                linkStr = useSsl ? ("tds://" + MQcProject + "." + MQcDomain + "." + MQcServer.Replace("https://", "") + "/TestRunsModule-00000000090859589?EntityType=IRun&EntityID=" + runId)
```

Comment:
What does the hardcoded string means?

Reply:
That is the link that opens the results in ALM